src/appliance-onboarding-script/appliance_setup/run.py

```diff
@@ -129,6 +130,11 @@ def _populate_default_values_of_optional_fields_in_config(config):
         logging.info("avs enabled")
         avs_config_validator = ConfigValidator(config)
         avs_config_validator.validate_avs_config()
+
+        if not config["applianceControlPlaneIpAddress"].strip():
```

Comment:
> > ```
> > def validate_static_ip_nw_config(self):
> > ```
> 
> Would the config validation succeed here for the new config? I think we are calling this before populating the config.
> 
> Refers to: src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/utils/user_config_validatator.py:9 in [e24e7d5](https://github.com/Azure/ArcOnAVS/commit/e24e7d5e287af9db5b38be213285387eeb329157). [](commit_id = [e24e7d5](https://github.com/Azure/ArcOnAVS/commit/e24e7d5e287af9db5b38be213285387eeb329157), deletion_comment = False)

I thought we will publish a single config file in public docs so that they won't get confused and customer will mention the values he may need. We will distinguish between both the config if value is present in applianceControlPlaneIpAddress or not.
option1:
{
  "subscriptionId": "",
  "resourceGroup": "",
  "applianceControlPlaneIpAddress": "",
  "privateCloud": "",
  "isStatic": true,
  "staticIpNetworkDetails": {
    "networkForApplianceVM": "",
    "networkCIDRForApplianceVM": "",
    "k8sNodeIPPoolStart": "",
    "k8sNodeIPPoolEnd": "",
    "gatewayIPAddress": ""
  }
}

option2:
{
  "subscriptionId": "",
  "resourceGroup": "",
  "applianceControlPlaneIpAddress": "leaves empty",
  "privateCloud": "",
  "isStatic": true,
  "staticIpNetworkDetails": {
    "networkForApplianceVM": "",
    "networkCIDRForApplianceVM": "",
    "k8sNodeIPPoolStart": "leaves empty",
    "k8sNodeIPPoolEnd": "leaves empty",
    "gatewayIPAddress": "leaves empty"
  }
}

Reply:
Please suggest what is expected of customer. which would be a better customer experience?

Reply:
I am just thinking this would be a consistent logic:

If all values are present, let's consider it as old config

if only ip cidr and segment name is given, let's consider it as new config

anything between. shall we error out a message of "Invalid config"

but we will have same config file for both configs and keys need to be present(though values are optional in new config)

Reply:
handled like this https://github.com/Azure/ArcOnAVS/pull/42#discussion_r1009711645